api/tests/test_audit_log.py

```diff
@@ -0,0 +1,176 @@
+    qs = AuditLogEntry.objects.filter(message__audit_event__target__path=url)
+    # There should be just 1 audit log entry
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+    # The audit log entry should have all the delivery log ids in it
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(log.pk) for log in delivery_logs
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.READ.value
+
+
+def test_creating_delivery_log_writes_create_log(admin_user):
+    """
+    Test that creating a DeliveryLog object writes a create log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    # Get the URL for creating a new DeliveryLog object
+    url = reverse("admin:api_deliverylog_add")
+
+    # Send a POST request with data to create the object
+    response = client.post(url, {"user": admin_user.pk, "report": "null"})
+
+    # Assuming successful creation redirects to changelist view
+    assert response.status_code == 302
+
+    # Find the newly created DeliveryLog object
+    delivery_log = DeliveryLog.objects.latest("pk")
+
+    # Assert that an AuditLogEntry was created for the creation
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.CREATE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.CREATE.value
+
+
+def test_updating_delivery_log_writes_update_log(admin_user):
+    """
+    Test that updating a DeliveryLog object writes an update log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    delivery_log = DeliveryLogFactory()
+
+    # Get the URL for updating the DeliveryLog object
+    url = reverse("admin:api_deliverylog_change", args=[delivery_log.pk])
+
+    # Send a POST request with updated data
+    updated_data = {
+        "user": admin_user.pk,
+        "report": "null",
+    }  # Example updated data
+    response = client.post(url, updated_data)
+    assert response.status_code == 302  # Redirect after successful update
+
+    # Assert that an AuditLogEntry was created for the update
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.UPDATE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
```

Comment:
Same comment here as in https://github.com/City-of-Helsinki/notification-service-api/pull/69/files#r1843403523
```suggestion
    assert sorted(
        audit_log_entry.message["audit_event"]["target"]["object_ids"]
    ) == sorted(str(delivery_log.pk))
```